internal/store/bigtable/group.go

```diff
@@ -34,64 +34,104 @@ var groupRank = map[string]int{
 
 const defaultRank = 9999
 
+// Table holds the bigtable name and client stub.
+type Table struct {
+	name  string
+	table *cbt.Table
+}
+
+// NewTable creates a new Table struct.
+func NewTable(name string, table *cbt.Table) *Table {
+	return &Table{name: name, table: table}
+}
+
+// Name access the name of a table
+func (t *Table) Name() string {
+	return t.name
+}
+
 // Group represents all the cloud bigtables that mixer talks to.
 type Group struct {
-	baseTables  []*cbt.Table
-	branchTable *cbt.Table
-	branchLock  sync.RWMutex
-	isProto     bool
+	tables          []*Table
+	lock            sync.RWMutex
+	branchTableName string
+	isProto         bool
 }
 
-//  TableConfig represents the config for a list bigtables.
+//  TableConfig represents the config for a list of Bigtable.
 type TableConfig struct {
 	Tables []string `json:"tables,omitempty"`
 }
 
 // NewGroup creates a BigtableGroup
 func NewGroup(
-	baseTables []*cbt.Table,
-	branchTable *cbt.Table,
+	tables []*Table,
+	branchTableName string,
 	useImportGroup bool,
 ) *Group {
+	if useImportGroup {
+		SortTables(tables)
+	}
 	return &Group{
-		baseTables:  baseTables,
-		branchTable: branchTable,
-		isProto:     useImportGroup,
+		tables:  tables,
+		isProto: useImportGroup,
 	}
 }
 
-// BaseTables is the accessor for base bigtables
-func (g *Group) BaseTables() []*cbt.Table {
-	return g.baseTables
+// Tables is the accessor for all the Bigtable client stubs.
+func (g *Group) Tables() []*cbt.Table {
+	g.lock.RLock()
+	defer g.lock.RUnlock()
+	result := []*cbt.Table{}
+	for _, t := range g.tables {
+		result = append(result, t.table)
+	}
+	return result
 }
 
-// BranchTable is the accessor for branch bigtable
-func (g *Group) BranchTable() *cbt.Table {
-	g.branchLock.RLock()
-	defer g.branchLock.RUnlock()
-	return g.branchTable
+// TableNames is the accesser to get all the Bigtable names.
+func (g *Group) TableNames() []string {
+	g.lock.RLock()
+	defer g.lock.RUnlock()
+	result := []string{}
+	for _, t := range g.tables {
+		result = append(result, t.name)
+	}
+	return result
 }
 
-// UpdateBranchTable updates the branch bigtable
-func (g *Group) UpdateBranchTable(branchTable *cbt.Table) {
-	g.branchLock.Lock()
-	defer g.branchLock.Unlock()
-	g.branchTable = branchTable
+// UpdateBranchTable updates the branch Bigtable.
+func (g *Group) UpdateBranchTable(branchTable *Table) {
+	g.lock.Lock()
+	defer g.lock.Unlock()
+	tables := []*Table{}
+	for _, t := range g.tables {
+		if t.name != g.branchTableName {
+			tables = append(tables, t)
+		}
+	}
+	tables = append(tables, branchTable)
+	g.branchTableName = branchTable.name
+	g.tables = tables
+	if g.isProto {
```

Comment:
else, is the assertion that branch cache always come after base?  Wonder if it's worth asserting?

Reply:
Done